test/functional/feature_maxuploadtarget.py

```diff
@@ -33,48 +37,97 @@ def on_block(self, message):
+        return tx
+
     def set_test_params(self):
         self.setup_clean_chain = True
-        self.num_nodes = 1
-        self.extra_args = [["-maxuploadtarget=800"]]
-
-        # Cache for utxos, as the listunspent may take a long time later in the test
-        self.utxo_cache = []
-
-    def run_test(self):
+        self.num_nodes = 3
         # Before we connect anything, we first set the time on the node
         # to be in the past, otherwise things break because the CNode
         # time counters can't be reset backward after initialization
-        old_time = int(time.time() - 2*60*60*24*7)
-        self.nodes[0].setmocktime(old_time)
+        self.mocktime =  int(time.time() - 2*60*60*24*7)
+        self.genesisBlock =  createTestGenesisBlock(self.signblockpubkey, self.signblockprivkey, self.mocktime)
 
+        self.utxo_cache = []
+
+    def setup_network(self):
+        self.add_nodes(3)
+        self.start_node(0, ["-maxuploadtarget=200"])
+        self.start_node(1)
+        self.start_node(2, ["-maxuploadtarget=800"])
+        connect_nodes_bi(self.nodes, 0,1)
+        connect_nodes_bi(self.nodes, 0,2)
+
+    def run_test(self):
         # Generate some old blocks
         self.nodes[0].generate(31, self.signblockprivkey_wif)
+        self.utxo_cache = self.nodes[0].listunspent()
+        self.sync_all()
 
-        # p2p_conns[0] will only request old blocks
-        # p2p_conns[1] will only request new blocks
-        # p2p_conns[2] will test resetting the counters
+        # p2p_conns[0] will only request old blocks and resetting the counters
+        # p2p_conns[1] will request new and old blocks
+        # p2p_conns[2] will test whitelist
         p2p_conns = []
 
-        for _ in range(3):
-            p2p_conns.append(self.nodes[0].add_p2p_connection(TestP2PConn(self.nodes[0].time_to_connect)))
+        p2p_conns.append(self.nodes[0].add_p2p_connection(TestP2PConn(self.nodes[0].time_to_connect)))
+        p2p_conns.append(self.nodes[1].add_p2p_connection(TestP2PConn(self.nodes[1].time_to_connect)))
+        p2p_conns.append(self.nodes[2].add_p2p_connection(TestP2PConn(self.nodes[2].time_to_connect)))
 
         # Now mine a big block
-        mine_large_block(self.nodes[0], self.utxo_cache)
+        big_old_block = self.mine_large_block(self.nodes[0], self.utxo_cache)
+        self.sync_all()
 
         # Store the hash; we'll request this later
-        big_old_block = self.nodes[0].getbestblockhash()
         old_block_size = self.nodes[0].getblock(big_old_block, True)['size']
         big_old_block = int(big_old_block, 16)
 
+        self.nodes[0].generate(100, self.signblockprivkey_wif)
+        self.sync_all()
+
         # Advance to two days ago
-        self.nodes[0].setmocktime(int(time.time()) - 2*60*60*24)
+        self.mocktime =  int(time.time()) - 2*60*60*24
+        self.nodes[0].setmocktime(self.mocktime)
+        self.nodes[1].setmocktime(self.mocktime)
+        self.nodes[2].setmocktime(self.mocktime)
```

Comment:
Bitcoin also does not seem to mock the time of nodes 1 and 2, but what is the intention of this change?

Reply:
When we use generate rpc, multiple blocks with the same timestamp skews the clock and later blocks are rejected because of their timestamp. mock was needed to. synchronise. the nodes too.

Reply:
Why hasn't it changed in Bitcoin?

Reply:
The clock skew again is tapyrus specific as mentioned in the other comment: https://github.com/chaintope/tapyrus-core/pull/297#discussion_r1541380127